quickwit-cli/src/main.rs

```diff
@@ -34,7 +33,7 @@ use quickwit_core::indexing::{index_data, IndexDataParams};
 
 struct CreateIndexArgs {
     index_uri: String,
-    doc_mapper_type: DocMapperType,
+    doc_mapper_type: String,
```

Comment:
I think that could be Box<dyn DocMapper>

Reply:
Not sure I understand?

Reply:
It could have been Box<dyn DocMapper> or Arc<dyn DocMapper> instead of string.

Reply:
To build the doc mapper we would need the json file though.

Reply:
So I still beleive  this might not be the right approach. this is the usage of create_index from a client perspective.
```rust
let doc_mapper = build_doc_mapper(&args.doc_mapper_type, mapper_config)?;
    let index_metadata = IndexMetadata {
        index_id: index_id.to_string(),
        index_uri: args.index_uri.to_string(),
        doc_mapper,
    };
    create_index(metastore_uri, index_metadata).await?;
```
as you can see what you are suggesting, we put it `IndexMetadata`. so a client to use the `create_index` from core needs to construct metastore_uri & index_metadata. I think we should not move the construction of the mapper in the cli parsing.

Reply:
@fulmicoton  ignore this I have a way forward now with @fmassot .